torchdata/datapipes/iter/util/_tfrecord_example_pb2.py

```diff
@@ -0,0 +1,698 @@
+# Generated by the protocol buffer compiler.  DO NOT EDIT!
```

Comment:
This file should be generated, not part of git repo. Instead please put the original protobuf file + generation script (inside of tests folder).

Reply:
I think we would need to include several protobuf files (example.proto depends on other files). Should the file be generated in setup.py?

Reply:
If there are only a few protobuf files, we can add them to the library without having them generated.

Do you know how many there are?

Reply:
I think there are two: "example.proto" and "feature.proto" https://github.com/tensorflow/tensorflow/tree/master/tensorflow/core/example

Reply:
But are you sure the "tests" folder is a good fit for ".proto" files?

Reply:
No, let's not put them in the `tests` folder, but somewhere within the main folder that actually get distributed. Let me take another holistic look at this PR and get back to you.

Reply:
@jkulhanek Can we generate this file once the first time the DataPipe is called (and just use it and not re-generate in the subsequent usages)?

In terms of location, I think `torchdata/datapipes/iter/util/protobuf_template` should be appropriate.

Reply:
This also means we would have to take care of locking (multiple threads/processes could be writing to the same file). This could likely happen in distributed training. Can't we generate it when shipping the package such that the generated file would be a part of it?

Reply:
In that case, putting it in `setup.py` is fine. Does the generation require extra dependencies?

Reply:
What about we raise an error message with instructions (and a script) to generate the file within the library if the file is missing? I think that avoids add the generation to `setup.py` and the multiprocessing issue.

cc: @VitalyFedyunin

Reply:
Can't we just leave the generated file in the package? It seems to me like the easiest option and we would not require users to install protobuf compiler before using the loader.

Reply:
Agree to leave it as is as soon as we place source .proto files besides it.

Reply:
Agree, we should add the generated files and the source .proto files as well. Somewhere like `torchdata/datapipes/iter/util/protobuf_template`.